src/core/Akka.Cluster.Tests.MultiNode/FailureDetectorPuppet.cs

```diff
@@ -69,6 +69,97 @@ public override void HeartBeat()
         {
             _status.CompareAndSet(Status.Unknown, Status.Up);
         }
+
```

Comment:
This test has used the old version with a value type, therefore I have copied the old version into this class.